### PR TITLE
feat: EIP-1271 accordion details

### LINKED ISSUE #1157: [EIP-1271] Accordion details
## What is the feature about

Implementation of EIP-1271 "Messages" details.

## The list of requirements

- Use stubbed `useMessages` hook that returns mocked data from the accordion PR
- Create `MsgDetails`, similar to `TxDetails`

Note: the "stepper" will be tackled in separate PRs, found in the epic [here](https://github.com/safe-global/web-core/issues/1088).

## Designs/sketches

image

## Links

- [Problem statement/data structure](https://www.notion.so/gnosis-safe/EIP-1271-signatures-can-only-be-generated-on-chain-2800b5e516124d959c38ba8f2042f9f7?d=e8ca1d332df54ae587e66240fd40e230#a8489e3724234f75a9c3d389e323a725)
- [Figma design](https://www.figma.com/file/x1uuIJsOZkyv5cwNlDdVC1/EIP-1271?node-id=1%3A2&t=F7CHK05WWFuPl1sq-0)

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #1157

## How this PR fixes it

`MsgDetails` now renders the details of a message within the `MsgSummary` accordion, matching that of the transaction list style.

## How to test it

Open the "Messages" from the transaction list and expand the accordions. Observe the stubbed message details rendered in the same style as multisend transactions.

## Screenshots

image

image

### CODE REVIEW

File: src/hooks/useMessages.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ export type Message = {
     owner: AddressEx
     signature: string
   }[]
+  proposedBy: AddressEx
```
Comment (iamacook) [inline src/hooks/useMessages.ts:35]: This has been subsequently added by infra.

File: src/components/messages/MsgDetails/index.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        <div className={txDetailsCss.multiSend}>
+          {msg.confirmations.map((confirmation, i) => (
+            <Accordion
+              variant="elevation"
+              key={confirmation.signature}
+              defaultExpanded={confirmation.owner.value === wallet?.address}
+            >
+              <AccordionSummary expandIcon={<ExpandMoreIcon />}>
+                <div className={singleTxDecodedCss.summary}>
+                  <CodeIcon />
```
Comment (schmanu) [inline src/components/messages/MsgDetails/index.tsx:93]: IMO we should use a different Icon here as it's not multisend calls / functions but confirmations.
Probably a checkmark or something like that.

**Review by schmanu — COMMENTED** [review]
I noticed some small differences to the figma designs. 

Otherwise it is looking good :)

File: src/components/messages/MsgDetails/index.tsx (hunk 2 above)
Comment (iamacook) [reply src/components/messages/MsgDetails/index.tsx:95]: The design uses a `CodeIcon` but I tend to agree. cc @liliiaorlenko

File: src/components/messages/MsgDetails/index.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    setShowMsg((prev) => !prev)
+  }
+
+  if (typeof message === 'string') {
+    return <>{message}</>
+  }
+
+  return (
+    <div>
+      {showMsg && (
```
[the remark this answers is not in the document]
Comment (iamacook) [reply src/components/messages/MsgDetails/index.tsx:32]: I discussed this and confirmed my implementation with @liliiaorlenko.

There are two slightly differing screens on Figma and [this is the implemented one](https://www.figma.com/file/x1uuIJsOZkyv5cwNlDdVC1/EIP-1271?node-id=253%3A4454&t=h6VdTB2z1Sn06Z2r-0). Semantically, `all` doesn't make sense in this context so I removed it.

File: src/components/messages/MsgDetails/index.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import { InfoDetails } from '@/components/transactions/InfoDetails'
+import { TxDataRow } from '@/components/transactions/TxDetails/Summary/TxDataRow'
+import type { Message } from '@/hooks/useMessages'
+
+import txDetailsCss from '@/components/transactions/TxDetails/styles.module.css'
+import singleTxDecodedCss from '@/components/transactions/TxDetails/TxData/DecodedData/SingleTxDecoded/styles.module.css'
+import infoDetailsCss from '@/components/transactions/InfoDetails/styles.module.css'
+import useWallet from '@/hooks/wallets/useWallet'
+
+const Msg = ({ message }: { message: Message['message'] }): ReactElement => {
```
[the remark this answers is not in the document]
Comment (iamacook) [reply src/components/messages/MsgDetails/index.tsx:19]: @liliiaorlenko agreed that we should later revisit the "prettifying" and add syntax highlight. I'd suggest we revisit it then.

**Review by schmanu — APPROVED** [review]
🚀 💯

**Pull request merged**